lms/static/js/groups/views/cohort_discussions_course_wide.js

```diff
@@ -29,13 +29,15 @@
                      * @returns {HtmlSnippet} - HTML list for course-wide discussion topics.
                      */
                     getCourseWideDiscussionsHtml: function(courseWideDiscussions) {
-                        var subCategoryTemplate = HtmlUtils.template($('#cohort-discussions-subcategory-tpl').html()),
+                        var entryTemplate = HtmlUtils.template($('#cohort-discussions-subcategory-tpl').html()),
```

Comment:
@muzaffaryousaf @attiyaIshaque: Thanks for your patience with this.  Hopefully we can help once we are all clear.

@muzaffaryousaf: Can you carefully read the following notes and determine if you think anything is inaccurate?
```
Course wide discussions:
- children are all entries (leaf nodes)

Inline discussions:
- contains top-level container named subcategories
  - children can be mix of deeper subcategories or entries (leaf nodes) (in any order?)
  
Other details:
- Entries always have an id and represent a discussion topic.
- Entries are always leaf nodes, and never have children.
- Categories and subcategories are also names used in the UI to differentiate between top-level versus nested categories in inline discussions. 
  - In the UI (and Read the Docs), top-level categories are actually represented as the top-level subcategories data-structure.
  - In the UI (and Read the Docs), either a category or a subcategory may actually be an entry (leaf node) in the data-structure.

The most confusing term is subcategory, because:
- A subcategory is sometimes a container of subcategories (everywhere).
- A subcategory is sometimes a top-level container (in the UI code and the data-structure, but not the UI or Read the Docs).
- A subcategory is sometimes an entry or topic (in the UI code, the UI, and Read the Docs, but not in the data-structure).
```

Unfortunately, it is out-of-scope to fix all of this confusion as part of this PR.  

Here's what I think we should do:
1. Once @muzaffaryousaf fixes or signs off on my comments above, I will try to document it in Confluence so it is captured somewhere.  I will also need to update the [Glossary in Confluence](https://openedx.atlassian.net/wiki/pages/viewpage.action?pageId=29688700) which is currently inaccurate.
2. I would leave the above rename to `entryTemplate` as-is, which matches the other usage.
3. I would add the following comment into `cohort-discussions-subcategory.underscore`:
```
<%
// TODO: This underscore template is a template for entries (leaf nodes), and not for subcategories.
// We should rename uses of the term subcategory in this template, since subcategory can mean too
// many different things.
%>
```
4. I would rename `cohort-discussions-subcategory.underscore` to `cohort-discussions-entry.underscore` if it doesn't cause much of a problem, since it seems to just be used in 2 places.

Let me know your thoughts.